drivers/sensor/qdec_nxp_s32/qdec_nxp_s32.c

```diff
@@ -0,0 +1,444 @@
+		config->trgmux_config->paxLogicTrigger[0]->Input,
+		config->trgmux_config->paxLogicTrigger[0]->Output);
+
+	if (Lcu_Ip_Init(config->lcu_config)) {
+		LOG_ERR("Could not initialize Lcu");
+		return -EINVAL;
+	}
+
+	/* Unmask relevant LCU OUT Channels */
+	Lcu_Ip_SyncOutputValueType EncLcuEnable[4U];
+
+	EncLcuEnable[0].LogicOutputId = LCU_LOGIC_OUTPUT_0;
+	EncLcuEnable[0].Value = 1U;
+	EncLcuEnable[1].LogicOutputId = LCU_LOGIC_OUTPUT_1;
+	EncLcuEnable[1].Value = 1U;
+	EncLcuEnable[2].LogicOutputId = LCU_LOGIC_OUTPUT_2;
+	EncLcuEnable[2].Value = 1U;
+	EncLcuEnable[3].LogicOutputId = LCU_LOGIC_OUTPUT_3;
+	EncLcuEnable[3].Value = 1U;
+	Lcu_Ip_SetSyncOutputEnable(EncLcuEnable, 4U);
+
+	emios_inst = config->emios_inst;
+	emios_hw_ch_cw =  config->emios_channels[EMIOS_CW_CH_IDX];
+	emios_hw_ch_ccw =  config->emios_channels[EMIOS_CCW_CH_IDX];
+
+	/* Initialize the positions of the eMios hw channels used for QDEC
+	 * to be beyond the eMios pwm hw channels. Currently only pwm and qdec
+	 * are using the eMios channels so qdec ones are the last two.
+	 */
+	eMios_Icu_Ip_IndexInChState[emios_inst][emios_hw_ch_cw]
+		= EMIOS_ICU_IP_NUM_OF_CHANNELS_USED - 2;
+	eMios_Icu_Ip_IndexInChState[emios_inst][emios_hw_ch_ccw]
+		= EMIOS_ICU_IP_NUM_OF_CHANNELS_USED - 1;
+
+	/* Set Overflow Notification for eMIOS channels meant
+	 * for Clockwise and Counterclock rotation counters
+	 */
+	eMios_Icu_Ip_ChState[eMios_Icu_Ip_IndexInChState[emios_inst][emios_hw_ch_cw]]
+		.eMiosOverflowNotification = config->emios_cw_overflow_cb;
+	eMios_Icu_Ip_ChState[eMios_Icu_Ip_IndexInChState[emios_inst][emios_hw_ch_ccw]]
+		.eMiosOverflowNotification = config->emios_ccw_overflow_cb;
+
+	Emios_Icu_Ip_SetInitialCounterValue(
+		config->emios_inst, config->emios_channels[EMIOS_CW_CH_IDX], (uint32_t)0x1U);
+	Emios_Icu_Ip_SetInitialCounterValue(
+		config->emios_inst, config->emios_channels[EMIOS_CCW_CH_IDX], (uint32_t)0x1U);
+
+	Emios_Icu_Ip_SetMaxCounterValue(config->emios_inst,
+		config->emios_channels[EMIOS_CW_CH_IDX],
+		EMIOS_ICU_IP_COUNTER_MASK);
+	Emios_Icu_Ip_SetMaxCounterValue(config->emios_inst,
+		config->emios_channels[EMIOS_CCW_CH_IDX],
+		EMIOS_ICU_IP_COUNTER_MASK);
+
+	/* This API sets MCB/EMIOS_ICU_MODE_EDGE_COUNTER mode */
+	Emios_Icu_Ip_EnableEdgeCount(config->emios_inst, config->emios_channels[EMIOS_CW_CH_IDX]);
+	Emios_Icu_Ip_EnableEdgeCount(config->emios_inst, config->emios_channels[EMIOS_CCW_CH_IDX]);
+
+	LOG_DBG("Init complete");
+
+	return 0;
+}
+
+#define EMIOS_NXP_S32_CALLBACK(n)								\
+	static uint32_t qdec##n##_emios_cw_overflow_count;					\
+	static uint32_t qdec##n##_emios_ccw_overflow_count;					\
+												\
+	static void qdec##n##_emios_overflow_count_cw_callback(void)				\
+	{											\
+		qdec##n##_emios_cw_overflow_count++;						\
+	}											\
+												\
+	static void qdec##n##_emios_overflow_count_ccw_callback(void)				\
+	{											\
+		qdec##n##_emios_ccw_overflow_count++;						\
+	}											\
```

Comment:
This is I meant in https://github.com/zephyrproject-rtos/zephyr/pull/65035#discussion_r1408479814:
```c
static void qdec_emios_overflow_count_cw_callback(const struct device *dev)
{
	struct qdec_s32_data *data = dev->data;

	data->emios_cw_overflow_count++;
}

#define EMIOS_NXP_S32_CALLBACK(n) \
static void qdec##n##_emios_overflow_count_cw_callback(void)
{ \
	qdec_emios_overflow_count_cw_callback(DEVICE_DT_INST_GET(n)); \
}
```

That way you don't need `SENSOR_DRIVER_API`  or `QDEC_NXP_S32_FETCH` macros, which neither make sense nor even work (you're back to duplicate symbols again). It doesn't make sense to have instance-specific implementations of sensor API functions that take `const struct device *` as an argument.

Reply:
Thanks @MaureenHelm for helping out..
I have made these changes